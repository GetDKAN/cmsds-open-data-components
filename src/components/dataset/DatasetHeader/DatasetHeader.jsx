import React from 'react'
import PropTypes from 'prop-types'
import Text from '../../common/Text/Text'
import DatasetDate from '../../common/DatasetDate/DatasetDate'
import DatasetHeaderLoading from '../DatasetHeaderLoading/DatasetHeaderLoading'

import './DatasetHeader.scss'
const DatasetHeader = ({ title, description, modified, released, refresh, isLoading }) => {
  const date = { modified, released, refresh }
  if (isLoading) {
    return <DatasetHeaderLoading />
  }
  return (
    <header className='dataset-header'>
      {title && <h1 className='dataset-title'>{title}</h1>}
      <div className='dataset-description'>
        <Text value={description} />
      </div>
      <DatasetDate date={date} updatedBoldLabel />
    </header>
  )
}

DatasetHeader.propTypes = {
  /**
   * Dataset title
   */
  title: PropTypes.string,
  /**
   * Dataset description
   */
  description: PropTypes.string,
  /**
   * Dataset modified date string
   */
  modified: PropTypes.string,
  /**
   * Dataset released date string
   */
  released: PropTypes.string,
  /**
   * Dataset anticipated refresh date string
   */
  refresh: PropTypes.string,
  /**
   * `true` shows the DatasetHeaderLoading component
   */
  isLoading: PropTypes.bool
}
DatasetHeader.displayName = 'DatasetHeader'
export default DatasetHeader
