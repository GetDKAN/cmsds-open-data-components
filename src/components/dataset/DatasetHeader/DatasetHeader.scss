.dataset-header {
  .dataset-title {
    word-wrap: break-word;
  }

  .dataset-description {
    margin-bottom: 16px;
  }
}
