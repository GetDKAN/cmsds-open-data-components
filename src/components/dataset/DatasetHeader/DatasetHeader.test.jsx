import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import DatasetHeader from './DatasetHeader'
import moment from 'moment'

const componentArgs = {
  title: 'Medicare Plan Info Data',
  description: 'A dataset showing all Medicare plan information from CMS.',
  modified: '2021-11-29',
  released: '2021-11-29',
  refresh: '2022-11-29',
  isLoading: false
}

const renderComponent = async (args = componentArgs) => render(
  <DatasetHeader {...args} />
)

describe('DatasetHeader component.', () => {
  it('Matches snapshot.', async () => {
    const renderedDatasetHeader = renderer.create(
      <DatasetHeader {...componentArgs} />
    ).toJSON()

    expect(renderedDatasetHeader).toMatchSnapshot()
  })

  it('Renders all passed prop values.', async () => {
    const { container } = await renderComponent()

    expect(screen.queryByText('Medicare Plan Info Data')).toBeInTheDocument()
    expect(screen.queryByText('A dataset showing all Medicare plan information from CMS.')).toBeInTheDocument()
    expect(container.querySelector('.dataset-date-item:first-child span.dataset-data-item-label').textContent).toBe(`Last Modified: ${moment(componentArgs.modified).format('MMMM D, YYYY')}`)
    expect(container.querySelector('.dataset-date-item:nth-child(3) span.dataset-data-item-label').textContent).toBe(`Released: ${moment(componentArgs.released).format('MMMM D, YYYY')}`)
    expect(container.querySelector('.dataset-date-item:last-child span.dataset-data-item-label').textContent).toBe(`Planned Update: ${moment(componentArgs.refresh).format('MMMM D, YYYY')}`)
  })

  it('Renders the loading component is \'isLoading\' prop is true.', () => {
    renderComponent({
      ...componentArgs,
      isLoading: true
    })

    expect(screen.queryByTestId('DatasetHeaderLoading')).toBeInTheDocument()
  })
})
