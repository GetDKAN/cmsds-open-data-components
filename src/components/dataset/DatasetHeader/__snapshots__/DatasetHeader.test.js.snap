// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetHeader component. Matches snapshot. 1`] = `
<header
  className="dataset-header"
>
  <h1
    className="dataset-title"
  >
    Medicare Plan Info Data
  </h1>
  <div
    className="dataset-description"
  >
    A dataset showing all Medicare plan information from CMS.
  </div>
  <div
    className="dataset-date"
  >
    <div
      className="dataset-date-item bold-label"
    >
      <span>
        Last Modified
      </span>
      : 
      November 29, 2021
    </div>
    <span
      className="bullet-point"
    >
      â€¢
    </span>
    <div
      className="dataset-date-item"
    >
      <span>
        Released
      </span>
      : 
      November 29, 2021
    </div>
  </div>
</header>
`;
